isobus/src/isobus_virtual_terminal_client_state_tracker.cpp

```diff
@@ -0,0 +1,116 @@
+//================================================================================================
+/// @file isobus_virtual_terminal_client_state_tracker.cpp
+///
+/// @brief A helper class to track the state of an active working set.
+/// @author Daan Steenbergen
+///
+/// @copyright 2023 The Open-Agriculture Developers
+//================================================================================================
+#include "isobus/isobus/isobus_virtual_terminal_client_state_tracker.hpp"
+
+#include "isobus/isobus/can_general_parameter_group_numbers.hpp"
+#include "isobus/isobus/can_network_manager.hpp"
+#include "isobus/isobus/can_stack_logger.hpp"
+#include "isobus/isobus/isobus_virtual_terminal_client.hpp"
+
+#include <algorithm>
+
+namespace isobus
+{
+	VirtualTerminalClientStateTracker::VirtualTerminalClientStateTracker(std::shared_ptr<ControlFunction> client) :
+	  client(client)
+	{
+	}
+
+	void VirtualTerminalClientStateTracker::initialize()
+	{
+		CANNetworkManager::CANNetwork.add_any_control_function_parameter_group_number_callback(static_cast<std::uint32_t>(CANLibParameterGroupNumber::VirtualTerminalToECU), process_rx_message, this);
```

Comment:
Probably want some way to remove this? Otherwise it could get called after this object is destructed?

Reply:
Oh yaeh definitely!